lib/mix/test/mix/project_test.exs

```diff
@@ -138,7 +138,12 @@ defmodule Mix.ProjectTest do
 
   defp assert_proj_dir_linked_or_copied(source, target, symlink_path) do
     case :file.read_link(source) do
-      {:ok, path} -> assert path == symlink_path
+      {:ok, path} -> 
+        case :os.type do
+          # relative symlink on Windows are broken, see symlink_or_copy/2
+          {:win32, _} -> assert path == Path.expand(Path.join([source, '../', symlink_path])) |> String.to_charlist
```

Comment:
We never use pipe once. So you can either make it all a pipeline `[source, '..', symlink_path] |> Path.join |> Path.expand |> String.to_charlist` or remove the pipe altogether. :)